lib/api/controllers/remoteActions/swagger.js

```diff
@@ -0,0 +1,145 @@
+var
+  fs = require('fs'),
+  _ = require('lodash'),
+  path = require('path'),
+  InternalError = require('kuzzle-common-objects').Errors.internalError,
+  q = require('q'),
+  stringify = require('json-stable-stringify'),
+  jsonToYaml = require('json2yaml'),  
+  generateSwagger,
+  writeFiles;
+
+generateSwagger = (routes, apiVersion) => {
+  var swagger;
+
+  swagger = {
+    basePath: '/api',
+    consumes: [
+      'application/json'
+    ],
+    host: 'sandbox.kuzzle.io:7511',
+    info: {
+      contact: {
+        email: 'hello@kuzzle.io',
+        name: 'Kuzzle team',
+        url: 'http://kuzzle.io'
+      },
+      description: 'The Kuzzle REST API',
+      license: {
+        name: 'Apache 2',
+        url: 'http://opensource.org/licenses/apache2.0'
+      },
+      title: 'Kuzzle API',
+      version: apiVersion
+    },
+    produces: [
+      'application/json'
+    ],
+    schemes: [
+      'http'
+    ],
+    swagger: '2.0',    
+    paths: {}
+  };
+
+  routes.forEach(route => {
+    var 
+      reg = /:([^\/]*)/g, 
+      m;
+
+    if (swagger.paths[route.url] === undefined) {
+      swagger.paths[route.url] = {};
+    }
+
+    if (swagger.paths[route.url][route.verb] === undefined) {
+      if (route.infos !== undefined) {
+        route.infos.description += '\nController: ' + route.controller + '. Action: ' + route.action + '.';
+        swagger.paths[route.url][route.verb] = route.infos;
+      } 
+      else {
+        // the route definision is minimal, lets create a default one
+        swagger.paths[route.url][route.verb] = {
+          description: 'Controller: ' + route.controller + '. Action: ' + route.action + '.',
+          produces: 'application/json',
+          responses: {
+            '200': {
+              description: 'OK',
+              schema: {
+                action: 'string',
+                controller: 'string',
+                error: 'object or null',
+                metadata: 'object',
+                requestId: 'string',
+                result: 'object',
+                scope: 'string or null',
+                state: 'string',
+                status: 'integer'
+              }
+            }
+          },
+          parameters: []
+        };
+      }
+
+      while ((m = reg.exec(route.url)) !== null) {
+        if (m.index === reg.lastIndex) {
```

Comment:
from: https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/RegExp/exec
> re 	lastIndex	The index at which to start the next match. When "g" is absent, this will remain as 0.